function/dataaccess/sam.js

```diff
@@ -13,42 +13,16 @@ const checkUserEnabled = async (appConfig, authToken) => {
   // query Sam for registration status + known email
   const reqOptions = {
     uri: `${opts.samUrl}/register/user/v2/self/info`,
-    headers: {
-      'Authorization': authToken,
-    },
-    json: true,
-    simple: false,
-    resolveWithFullResponse: true,
-    timeout: opts.samTimeout || 60000,
+    successCodes: [200],
+    timeout: opts.timeout,
   };
 
-  // TODO: move all the error-handling and REST-call mechanics
-  //  into a shared function, once we are making REST calls to
-  //  some other locations/from other files.
-  return await rp(reqOptions)
+  return await restclient.safeRequest(reqOptions, authToken)
       .then((response) => {
-        if (response.statusCode === 200) {
-          if (response.body.enabled) {
-            return response.body.userEmail;
-          } else {
-            throwHttpError(403, 'User is disabled.');
-          }
+        if (response.body.enabled) {
+          return response.body.userEmail;
         } else {
-          throwHttpError(response.statusCode,
-              (response.body.message ? response.body.message : response.statusCode));
-        }
-      })
-      .catch((err) => {
-        // if we have an HttpError from a disabled user or non-200 from Sam, rethrow it
-        if (err.name === 'HttpError') {
-          throw err;
-        } else {
-          // if request to Sam timed out, give a friendly error
-          if (err.name === 'RequestError' && err.cause.code === 'ESOCKETTIMEDOUT') {
-            throwHttpError(500, `Connection to ${reqOptions.uri} timed out.`);
-          } else {
-            throwHttpError(500, `Unexpected error: ${JSON.stringify(err)}`);
-          }
+          throwHttpError(403, 'User is disabled.');
```

Comment:
our proxies return 401 if user is disabled (I am not taking a stand on what is right or wrong, only that 403 is inconsistent)

Reply:
ah, good call! Though based on other discussion I may remove this.

Reply:
yeah, I removed it entirely. back to you for re-review, @dvoet